### PR TITLE
Add Google App Engine Provider To Docs

### PR DESCRIPTION
**Pull request by waprin**
Creates docs for Google App Engine deployment provider.

### CODE REVIEW

File: user/deployment/google-app-engine.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    deploy:
+      provider: gae
+      keyfile: "YOUR SERVICE ACCOUNT JSON FILE"
+      default: true
+      project: "YOUR PROJECT ID"
+
+You can create a Service Account by going to the [Google Cloud Console](console.developers.google.com), go to "APIs & auth" -> "Credentials",
+then click "Add Credential" and "Service Account", finally clicking "JSON" to downloa dthe JSON key.
+
+You must then encrypt your key before committing it to a repo. First make sure you have the Travis command line tool installed.
```
Comment (BanzaiMan) [inline user/deployment/google-app-engine.md]: It is technically incorrect that you "must". But it is a *_very_* good idea.

Please add a link to http://docs.travis-ci.com/user/encrypting-files/, too.

Comment (waprin) [inline user/deployment/google-app-engine.md]: Changed to strongly recommended.

Comment (waprin) [inline user/deployment/google-app-engine.md]: And added link.

**Comment by waprin** [comment]
Also @BanzaiMan I am planning on mentioning this provider on http://googlecloudplatform.blogspot.com/. l wanted to add a footnote like "Thanks to Lorenz Leutgeb for updating the App Engine provider, and Hiro Asari and the Travis team for their help reviewing and merging the feature." Of course I will not mention you if you object to that.

**Comment by BanzaiMan** [comment]
Hi, @waprin, We are more than happy to be mentioned! Thank you.

**Pull request merged**